src/main/java/com/pscheduler/server/model/CourseGPA.java

```diff
@@ -0,0 +1,176 @@
+package com.pscheduler.server.model;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Version;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+@Entity
+public class CourseGPA {
+
+    @Id
+    @Min(value = 10000)
+    @Max(value = 99999)
+    private int crn;
+
+    @NotNull
+    @Size(min = 2, max = 6)
+    private String subject;
+
+    @NotNull
+    @Size(min = 4, max = 5)
+    private String courseNumber;
+
+    @NotNull
+    @Size(min = 1, max = 100)
+    private String name;
+
+    @NotNull
+    @Size(min = 1, max = 50)
+    private String instructor;
+
+    @Min(value = 0)
+    @Max(value = 50)
+    private int credits;
+
+    @NotNull
+    @Min(value = 0)
+    @Max(value = 4)
+    private double gpa;
+
+    @NotNull
+    @Min(value = 0)
+    @Max(value = 1000)
```

Comment:
I've seen a class be >1000 capacity so lets make this 10000 to be safe

Reply:
Fixed